CalibTracker/SiStripHitEfficiency/plugins/HitEff.cc

```diff
@@ -369,15 +377,55 @@ void HitEff::analyze(const edm::Event& e, const edm::EventSetup& es) {
 
       // Check whether the trajectory has some missing hits
       bool hasMissingHits = false;
-      for (itm = TMeas.begin(); itm != TMeas.end(); itm++) {
-        auto theHit = (*itm).recHit();
+      unsigned int previous_layer = 999;
+      vector<unsigned int> missedLayers;
+
+      for (const auto& itm : TMeas) {
+        auto theHit = itm.recHit();
+        unsigned int iidd = theHit->geographicalId().rawId();
+        unsigned int layer = ::checkLayer(iidd, tTopo);
+        int missedLayer = int(layer + 1);
+        int diffPreviousLayer = int(layer - previous_layer);
+        if (doMissingHitsRecovery_) {
+          //Layers from TIB + TOB
+          if (diffPreviousLayer == -2 && missedLayer > k_LayersStart && missedLayer < k_LayersAtTOBEnd) {
+            missHitPerLayer[missedLayer] += 1;
+            hasMissingHits = true;
+          }
+          //Layers from TID
+          if (diffPreviousLayer == -2 && (missedLayer == 12 || missedLayer == 13)) {
+            missHitPerLayer[missedLayer] += 1;
+            hasMissingHits = true;
+          }
+          //Layers from TEC
+          if (diffPreviousLayer == -2 && missedLayer > k_LayersAtTIDEnd && missedLayer < k_LayersAtTECEnd) {
+            missHitPerLayer[missedLayer] += 1;
+            hasMissingHits = true;
+          }
+
+          //##### TID Layer 11 (transition TID -> TIB)
+          if ((int(layer) > k_LayersStart && int(layer) <= k_LayersAtTIBEnd) && (previous_layer == 12)) {
```

Comment:
Previous conditions were on `missedLayer,` these ones one `layer`: as a reviewer it is hard to follow, I imagine the same could be for a future developer who did not participate to this update to the code...

Reply:
I thought it would be better since the goal is to know if given layer is in TIB, while previous hit was before-last TID layer. changing it to "missedLayer" would add confusion would it not ?